cli/src/main/java/software/amazon/smithy/java/cli/SmithyCall.java

```diff
@@ -0,0 +1,253 @@
+                    }
+                    builder.putConfig(RegionSetting.REGION, awsRegion)
+                            .putSupportedAuthSchemes(new SigV4AuthScheme(defaultArnNamespace))
+                            .authSchemeResolver(AuthSchemeResolver.DEFAULT)
+                            .addIdentityResolver(new EnvironmentVariableIdentityResolver());
+                    break;
+                case "none":
+                    builder.authSchemeResolver(AuthSchemeResolver.NO_AUTH);
+                    break;
+                default:
+                    throw new IllegalArgumentException("Unsupported auth type: " + authType);
+            }
+        } else {
+            builder.authSchemeResolver(AuthSchemeResolver.NO_AUTH);
+        }
+    }
+
+    private void configureProtocol(DynamicClient.Builder builder, ShapeId serviceInput) {
+        if (protocol != null) {
+            switch (protocol.toLowerCase()) {
+                case AWS_JSON:
+                    builder.protocol(new AwsJson1Protocol(serviceInput));
+                    break;
+                case RPC_V2_CBOR:
+                    builder.protocol(new RpcV2CborProtocol(serviceInput));
+                    break;
+                case REST_JSON:
+                    builder.protocol(new RestJsonClientProtocol(serviceInput));
+                    break;
+                case REST_XML:
+                    builder.protocol(new RestXmlClientProtocol(serviceInput));
+                    break;
+                default:
+                    throw new IllegalArgumentException("Unsupported protocol type: " + protocol);
+            }
+        }
+    }
+
+    private void configureInputInterceptor(DynamicClient.Builder builder) {
+        if (input != null || inputPath != null) {
+            builder.addInterceptor(new ClientInterceptor() {
+                @Override
+                public void readBeforeTransmit(RequestHook<?, ?, ?> hook) {}
+            });
+        }
+    }
+
+    private Object executeClientCall(DynamicClient client) throws Exception {
+        if (input != null && inputPath != null) {
+            throw new IllegalArgumentException("Cannot specify both '--input-json' and '--input-path'. Please provide only one.");
+        }
+
+        if (input == null && inputPath == null) {
+            return client.call(operation).asObject();
+        }
+
+        Document inputDocument;
+        if (input != null) {
+            inputDocument = CODEC.createDeserializer(input.getBytes(StandardCharsets.UTF_8)).readDocument();
+        } else {
+            String content = Files.readString(Path.of(inputPath), StandardCharsets.UTF_8);
+            inputDocument = CODEC.createDeserializer(content.getBytes(StandardCharsets.UTF_8)).readDocument();
+        }
+
+        return client.call(operation, inputDocument).asObject();
```

Comment:
I'd return the document as-is, then serialize the document using CODEC for output. That way people can parse it and use it with jq, jmespath, etc.